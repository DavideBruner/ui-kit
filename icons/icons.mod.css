@import 'ui-kit/materials/colors.mod.css';

.theme-black {
  fill: var(--color-black);
}

.theme-grey {
  fill: var(--color-greyNevada);
}

.theme-white {
  fill: var(--color-white);
}

.theme-blue {
  fill: var(--color-blue);
}

.theme-green {
  fill: var(--color-primaryGreen);
}

.theme-apple-green {
  fill: var(--color-green);
}

.theme-orange {
  fill: var(--color-CTOrange);
}

.theme-red {
  fill: var(--color-red);
}

svg.scale:not(:root) {
  width: 100%;
  height: auto;
}
