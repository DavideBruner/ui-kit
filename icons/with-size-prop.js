import { mapProps } from 'recompose';
import classnames from 'classnames';
import styles from './icons.mod.css';

export default mapProps(({ size, ...remainingProps }) => ({
  ...remainingProps,
  ...(size === 'scale'
    ? { className: classnames(remainingProps.className, styles.scale) }
    : {}),
  width: size === 'small' ? 16 : 24,
  height: size === 'small' ? 16 : 24,
}));
