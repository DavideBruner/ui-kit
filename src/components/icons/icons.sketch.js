import React from 'react';
import styled from '@emotion/styled';
import * as UIKit from 'ui-kit';
import { Suite, SketchSpec } from '../../../test/percy';

const Inline = styled.div`
  display: flex;
  flex-direction: row;
`;

const icons = Object.keys(UIKit).filter(thing => thing.endsWith('Icon'));

const sizes = ['small', 'medium', 'big', 'scale'];
const themes = [
  'black',
  'grey',
  'white',
  'blue',
  'green',
  'green-light',
  'orange',
  'red',
];

export default (
  <Suite>
    <SketchSpec label="Icons">
      {icons.map(iconName => {
        const Icon = UIKit[iconName];
        return sizes.map(size => (
          <div key={`${iconName}-${size}`}>
            <Inline>
              {themes.map(theme => (
                <div
                  key={theme}
                  data-sketch-symbol={`Icon/${iconName}/${size}/${theme}`}
                >
                  <Icon size={size} theme={theme} />
                </div>
              ))}
            </Inline>
          </div>
        ));
      })}
    </SketchSpec>
  </Suite>
);
