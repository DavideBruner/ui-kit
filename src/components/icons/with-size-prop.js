import { mapProps } from 'recompose';
import classnames from 'classnames';
import styles from './icons.mod.css';

const ICON_SIZES = {
  SMALL: 12,
  MEDIUM: 16,
  BIG: 24,
};

const getIconSize = size => {
  switch (size) {
    case 'small':
      return ICON_SIZES.SMALL;
    case 'medium':
      return ICON_SIZES.MEDIUM;
    case 'big':
      return ICON_SIZES.BIG;
    default:
      return ICON_SIZES.BIG;
  }
};

export default mapProps(({ size, ...remainingProps }) => ({
  ...remainingProps,
  ...(size === 'scale'
    ? { className: classnames(remainingProps.className, styles.scale) }
    : {}),
  width: getIconSize(size),
  height: getIconSize(size),
}));
