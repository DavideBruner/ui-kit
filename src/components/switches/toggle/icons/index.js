import ToggleButtonIcon from './toggle-button-default.svg';
import ToggleBarIcon from './toggle-bar-default.svg';

export default {
  ButtonDefault: ToggleButtonIcon,
  BarDefault: ToggleBarIcon,
};
