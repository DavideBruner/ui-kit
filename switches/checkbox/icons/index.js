// NOTE: explicit names for component names
import CheckboxDefaultIcon from './checkbox-default.svg';
import CheckboxDefaultHoverIcon from './checkbox-hover.svg';
import CheckboxDefaultErrorIcon from './checkbox-error.svg';
import CheckboxCheckedIcon from './checkbox-checked.svg';
import CheckboxCheckedHoverIcon from './checkbox-checked-hover.svg';
import CheckboxCheckedErrorIcon from './checkbox-checked-error.svg';
import CheckboxDisabledIcon from './checkbox-disabled.svg';
import CheckboxCheckedDisabledIcon from './checkbox-checked-disabled.svg';
import CheckboxIndeterminateIcon from './checkbox-indeterminate.svg';
import CheckboxIndeterminateHoverIcon from './checkbox-indeterminate-hover.svg';
import CheckboxIndeterminateDisabledIcon from './checkbox-indeterminate-disabled.svg';
import CheckboxIndeterminateErrorIcon from './checkbox-indeterminate-error.svg';

export default {
  Default: CheckboxDefaultIcon,
  Hover: CheckboxDefaultHoverIcon,
  Error: CheckboxDefaultErrorIcon,
  Checked: CheckboxCheckedIcon,
  CheckedHover: CheckboxCheckedHoverIcon,
  CheckedError: CheckboxCheckedErrorIcon,
  Disabled: CheckboxDisabledIcon,
  CheckedDisabled: CheckboxCheckedDisabledIcon,
  Indeterminate: CheckboxIndeterminateIcon,
  IndeterminateHover: CheckboxIndeterminateHoverIcon,
  IndeterminateDisabled: CheckboxIndeterminateDisabledIcon,
  IndeterminateError: CheckboxIndeterminateErrorIcon,
};
