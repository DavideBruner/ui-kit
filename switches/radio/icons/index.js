// NOTE: explicit names for component names
import RadioOptionDefaultIcon from './radio-option-default.svg';
import RadioOptionDefaultHoverIcon from './radio-option-hover.svg';
import RadioOptionCheckedIcon from './radio-option-checked.svg';
import RadioOptionCheckedHoverIcon from './radio-option-checked-hover.svg';
import RadioOptionDisabledIcon from './radio-option-disabled.svg';
import RadioOptionCheckedDisabledIcon from './radio-option-checked-disabled.svg';

export default {
  Default: RadioOptionDefaultIcon,
  Hover: RadioOptionDefaultHoverIcon,
  Checked: RadioOptionCheckedIcon,
  CheckedHover: RadioOptionCheckedHoverIcon,
  Disabled: RadioOptionDisabledIcon,
  CheckedDisabled: RadioOptionCheckedDisabledIcon,
};
