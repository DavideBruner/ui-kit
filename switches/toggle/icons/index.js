import withSizeProp from '../with-size-prop';
import ToggleBackgroundDefaultIcon from './toggle-normal-bkgr.svg';
import ToggleBackgroundHoverIcon from './toggle-active-hover-bkgr.svg';
import ToggleBackgroundDisabledIcon from './toggle-disable-bkgr.svg';
import ToggleBackgroundCheckedIcon from './toggled-bkgr.svg';
import ToggleBackgroundCheckedHoverIcon from './toggled-active-hover-bkgr.svg';
import ToggleBackgroundCheckedDisabledIcon from './toggled-disable-bkgr.svg';
import ToggleButtonDefaultIcon from './toggle-normal-bt.svg';
import ToggleButtonHoverIcon from './toggle-active-hover-bt.svg';
import ToggleButtonDisabledIcon from './toggle-disable-bt.svg';
import ToggleButtonCheckedIcon from './toggled-bt.svg';
import ToggleButtonCheckedHoverIcon from './toggled-active-hover-bt.svg';
import ToggleButtonCheckedDisabledIcon from './toggled-disable-bt.svg';

export default {
  BackgroundDefault: ToggleBackgroundDefaultIcon,
  BackgroundHover: ToggleBackgroundHoverIcon,
  BackgroundDisabled: ToggleBackgroundDisabledIcon,
  BackgroundChecked: ToggleBackgroundCheckedIcon,
  BackgroundCheckedHover: ToggleBackgroundCheckedHoverIcon,
  BackgroundCheckedDisabled: ToggleBackgroundCheckedDisabledIcon,
  ButtonDefault: withSizeProp(ToggleButtonDefaultIcon),
  ButtonHover: withSizeProp(ToggleButtonHoverIcon),
  ButtonDisabled: withSizeProp(ToggleButtonDisabledIcon),
  ButtonChecked: withSizeProp(ToggleButtonCheckedIcon),
  ButtonCheckedHover: withSizeProp(ToggleButtonCheckedHoverIcon),
  ButtonCheckedDisabled: withSizeProp(ToggleButtonCheckedDisabledIcon),
};
