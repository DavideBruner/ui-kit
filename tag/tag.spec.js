import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router';
import AccessibleButton from '../buttons/accessible-button';
import { CloseBoldIcon } from '../icons';
import Tag, { TagNormalBody, TagLinkBody } from './tag';

describe('`<TagLinkBody />`', () => {
  const createTestProps = custom => ({
    type: 'normal',
    onClick: jest.fn(),
    linkTo: 'foo-too',
    isDisabled: false,
    children: 'Icecream we scream!',
    ...custom,
  });
  let props;
  let wrapper;

  describe('rendering', () => {
    beforeEach(() => {
      props = createTestProps();

      wrapper = shallow(<TagLinkBody {...props} />);
    });

    it('should output correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('when enabled (`isDisabled`)', () => {
      it('should supply `to` to `Link`', () => {
        expect(wrapper.find(Link)).toHaveProp('to', props.linkTo);
      });

      it('should apply clickable class name', () => {
        expect(wrapper.find(Link)).toHaveClassName('clickableContentWrapper');
      });

      it('should apply link class name', () => {
        expect(wrapper.find(Link)).toHaveClassName('plainLink');
      });
    });

    describe('when disabled (`isDisabled`)', () => {
      beforeEach(() => {
        props = createTestProps({ isDisabled: true });

        wrapper = shallow(<TagLinkBody {...props} />);
      });

      it('should not supply `to` to `Link`', () => {
        expect(wrapper.find(Link)).toHaveProp('to', undefined);
      });

      it('should apply disabled class name', () => {
        expect(wrapper.find(Link)).toHaveClassName('disabledContent');
      });
    });

    describe('when removeable (`onRemove`)', () => {
      beforeEach(() => {
        props = createTestProps({ onRemove: jest.fn() });

        wrapper = shallow(<TagLinkBody {...props} />);
      });

      it('should apply removeable class name', () => {
        expect(wrapper.find(Link)).toHaveClassName('removableContent');
      });
    });
  });

  describe('callbacks', () => {
    beforeEach(() => {
      props = createTestProps();

      wrapper = shallow(<TagLinkBody {...props} />);
    });

    describe('when enabled (`isDisabled`)', () => {
      describe('of `<Link />`', () => {
        describe('`onClick`', () => {
          beforeEach(() => {
            wrapper.find('Link').prop('onClick')();
          });

          it('should invoke `onClick`', () => {
            expect(props.onClick).toHaveBeenCalled();
          });
        });
      });
    });
  });
});

describe('`<TagNormalBody />`', () => {
  const createTestProps = custom => ({
    type: 'normal',
    onClick: jest.fn(),
    isDisabled: false,
    children: 'Icecream we scream!',
    ...custom,
  });
  let props;
  let wrapper;

  describe('rendering', () => {
    beforeEach(() => {
      props = createTestProps();

      wrapper = shallow(<TagNormalBody {...props} />);
    });

    it('should output correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('when enabled (`isDisabled`)', () => {
      it('should render an `AccessibleButton`', () => {
        expect(wrapper).toRender('AccessibleButton');
      });

      it('should supply `onClick` to `AccessibleButton`', () => {
        expect(wrapper.find('AccessibleButton')).toHaveProp(
          'onClick',
          props.onClick
        );
      });

      describe('when clickable (`onClick`)', () => {
        beforeEach(() => {
          props = createTestProps({ onClick: jest.fn() });

          wrapper = shallow(<TagNormalBody {...props} />);
        });

        it('should apply clickable class name', () => {
          expect(wrapper.find(AccessibleButton)).toHaveClassName(
            'clickableContentWrapper'
          );
        });
      });
    });

    describe('when disabled (`isDisabled`)', () => {
      beforeEach(() => {
        props = createTestProps({ isDisabled: true });

        wrapper = shallow(<TagNormalBody {...props} />);
      });

      it('should apply disabled class name', () => {
        expect(wrapper.find('AccessibleButton')).toHaveClassName(
          'disabledContent'
        );
      });
    });

    describe('when removable (`onRemove`)', () => {
      beforeEach(() => {
        props = createTestProps({ onRemove: jest.fn() });

        wrapper = shallow(<TagNormalBody {...props} />);
      });

      it('should apply removable class name', () => {
        expect(wrapper.find(AccessibleButton)).toHaveClassName(
          'removableContent'
        );
      });
    });
  });

  describe('callbacks', () => {
    beforeEach(() => {
      props = createTestProps();

      wrapper = shallow(<TagNormalBody {...props} />);
    });

    describe('when enabled (`isDisabled`)', () => {
      describe('of `<AccessibleButton />`', () => {
        describe('`onClick`', () => {
          beforeEach(() => {
            wrapper.find('AccessibleButton').prop('onClick')();
          });

          it('should invoke `onClick`', () => {
            expect(props.onClick).toHaveBeenCalled();
          });
        });
      });
    });
  });
});

describe('<Tag />', () => {
  const createTestProps = custom => ({
    children: 'Icecream we scream!',
    ...custom,
  });

  describe('rendering', () => {
    let props;
    let wrapper;

    describe('when of type `normal` (default)', () => {
      beforeEach(() => {
        props = createTestProps();

        wrapper = shallow(<Tag {...props} />);
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should apply `wrapperTypeNormal` class name', () => {
        expect(wrapper).toHaveClassName('wrapperTypeNormal');
      });
    });

    describe('when of type `warning`', () => {
      beforeEach(() => {
        props = createTestProps({ type: 'warning' });

        wrapper = shallow(<Tag {...props} />);
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should apply `wrapperTypeWarning` class name', () => {
        expect(wrapper).toHaveClassName('wrapperTypeWarning');
      });
    });

    describe('when disabled', () => {
      beforeEach(() => {
        props = createTestProps({
          isDisabled: true,
        });

        wrapper = shallow(<Tag {...props} />);
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should apply `disabledWrapper` class name', () => {
        expect(wrapper).toHaveClassName('disabledWrapper');
      });
    });

    describe('when removable (`onRemove`)', () => {
      beforeEach(() => {
        props = createTestProps({
          onRemove: jest.fn(),
        });

        wrapper = shallow(<Tag {...props} />);
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should render `CloseBoldIcon`', () => {
        expect(wrapper).toRender(CloseBoldIcon);
      });
    });

    describe('when `linkTo` is set', () => {
      beforeEach(() => {
        props = createTestProps({
          linkTo: '/foo',
          onClick: jest.fn(),
          isDisabled: false,
        });

        wrapper = shallow(<Tag {...props} />);
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should render a `<TagLinkBody />`', () => {
        expect(wrapper).toRender(TagLinkBody);
      });

      it('should supply `linkTo` to `<TagLinkBody />`', () => {
        expect(wrapper.find(TagLinkBody)).toHaveProp('linkTo', props.linkTo);
      });

      it('should supply `onClick` to `<TagLinkBody />`', () => {
        expect(wrapper.find(TagLinkBody)).toHaveProp('onClick', props.onClick);
      });

      it('should supply `onRemove` to `<TagLinkBody />`', () => {
        expect(wrapper.find(TagLinkBody)).toHaveProp(
          'onRemove',
          props.onRemove
        );
      });

      it('should supply `isDisabled` to `<TagLinkBody />`', () => {
        expect(wrapper.find(TagLinkBody)).toHaveProp(
          'isDisabled',
          props.isDisabled
        );
      });
    });

    describe('when `linkTo` is not set', () => {
      beforeEach(() => {
        props = createTestProps({
          onClick: jest.fn(),
          onRemove: jest.fn(),
          isDisabled: false,
        });

        wrapper = shallow(<Tag {...props} />);
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should render a `<TagNormalBody />`', () => {
        expect(wrapper).toRender(TagNormalBody);
      });

      it('should supply `onRemove` to `<TagNormalBody />`', () => {
        expect(wrapper.find(TagNormalBody)).toHaveProp(
          'onRemove',
          props.onRemove
        );
      });

      it('should supply `onClick` to `<TagNormalBody />`', () => {
        expect(wrapper.find(TagNormalBody)).toHaveProp(
          'onClick',
          props.onClick
        );
      });

      it('should supply `isDisabled` to `<TagNormalBody />`', () => {
        expect(wrapper.find(TagNormalBody)).toHaveProp(
          'isDisabled',
          props.isDisabled
        );
      });
    });
  });

  describe('callbacks', () => {
    let props;
    let wrapper;

    describe('of <AccessibleButton />', () => {
      describe('when enabled (`isDisabled`)', () => {
        describe('onClick', () => {
          beforeEach(() => {
            props = createTestProps({ onRemove: jest.fn() });

            wrapper = shallow(<Tag {...props} />);

            wrapper.find(AccessibleButton).prop('onClick')();
          });

          it('should invoke `onRemove`', () => {
            expect(props.onRemove).toHaveBeenCalled();
          });
        });
      });
    });
  });
});
