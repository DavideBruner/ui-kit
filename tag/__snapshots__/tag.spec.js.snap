// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> rendering when \`linkTo\` is not set should output correct tree 1`] = `
<div
  className="wrapper wrapperTypeNormal"
>
  <TagNormalBody
    isDisabled={false}
    onClick={[Function]}
    onRemove={[Function]}
    type="normal"
  >
    Icecream we scream!
  </TagNormalBody>
  <AccessibleButton
    buttonAttributes={Object {}}
    className="removeWrapper removeWrapperTypeNormal"
    isDisabled={false}
    isToggleButton={false}
    label="Remove"
    onClick={[Function]}
  >
    <CloseBoldIcon
      size="small"
    />
  </AccessibleButton>
</div>
`;

exports[`<Tag /> rendering when \`linkTo\` is set should output correct tree 1`] = `
<div
  className="wrapper wrapperTypeNormal"
>
  <TagLinkBody
    isDisabled={false}
    linkTo="/foo"
    onClick={[Function]}
    type="normal"
  >
    Icecream we scream!
  </TagLinkBody>
</div>
`;

exports[`<Tag /> rendering when disabled should output correct tree 1`] = `
<div
  className="wrapper wrapperTypeNormal disabledWrapper"
>
  <TagNormalBody
    isDisabled={true}
    type="normal"
  >
    Icecream we scream!
  </TagNormalBody>
</div>
`;

exports[`<Tag /> rendering when of type \`normal\` (default) should output correct tree 1`] = `
<div
  className="wrapper wrapperTypeNormal"
>
  <TagNormalBody
    isDisabled={false}
    type="normal"
  >
    Icecream we scream!
  </TagNormalBody>
</div>
`;

exports[`<Tag /> rendering when of type \`warning\` should output correct tree 1`] = `
<div
  className="wrapper wrapperTypeWarning"
>
  <TagNormalBody
    isDisabled={false}
    type="warning"
  >
    Icecream we scream!
  </TagNormalBody>
</div>
`;

exports[`<Tag /> rendering when removable (\`onRemove\`) should output correct tree 1`] = `
<div
  className="wrapper wrapperTypeNormal"
>
  <TagNormalBody
    isDisabled={false}
    onRemove={[Function]}
    type="normal"
  >
    Icecream we scream!
  </TagNormalBody>
  <AccessibleButton
    buttonAttributes={Object {}}
    className="removeWrapper removeWrapperTypeNormal"
    isDisabled={false}
    isToggleButton={false}
    label="Remove"
    onClick={[Function]}
  >
    <CloseBoldIcon
      size="small"
    />
  </AccessibleButton>
</div>
`;

exports[`\`<TagLinkBody />\` rendering should output correct tree 1`] = `
<Link
  className="contentWrapper contentWrapperNormal clickableContentWrapperNormal clickableContentWrapper plainLink"
  onClick={[Function]}
  onlyActiveOnIndex={false}
  style={Object {}}
  to="foo-too"
>
  <TextDetail>
    Icecream we scream!
  </TextDetail>
</Link>
`;

exports[`\`<TagNormalBody />\` rendering should output correct tree 1`] = `
<AccessibleButton
  buttonAttributes={Object {}}
  className="contentWrapper contentWrapperNormal clickableContentWrapperNormal clickableContentWrapper"
  isDisabled={false}
  isToggleButton={false}
  label="Click"
  onClick={[Function]}
>
  <TextDetail>
    Icecream we scream!
  </TextDetail>
</AccessibleButton>
`;
